packages/x-components/src/x-modules/tagging/service/__tests__/pdpd-add-to-cart.service.spec.ts

```diff
@@ -0,0 +1,124 @@
+import { InMemoryStorageService } from '@empathyco/x-storage-service';
+import { Store } from 'vuex';
+import { DefaultPDPAddToCartService } from '../pdp-add-to-cart.service';
+import { XPlugin } from '../../../../plugins/index';
+import { installNewXPlugin } from '../../../../__tests__/utils';
+import { taggingXModule } from '../../x-module';
+import { RootXStoreState } from '../../../../store/index';
+import { createResultStub } from '../../../../__stubs__/index';
+
+interface PDPdAddToCartServiceTestAPI {
+  service: DefaultPDPAddToCartService;
+  localStorageService: InMemoryStorageService;
+  sessionStorageService: InMemoryStorageService;
+  store: Store<RootXStoreState>;
+}
+
+function preparePDPAddToCartService(): PDPdAddToCartServiceTestAPI {
+  XPlugin.resetInstance();
+  installNewXPlugin();
+  XPlugin.registerXModule(taggingXModule);
+
+  const store = XPlugin.store;
+  const localStorageService = new InMemoryStorageService();
+  const sessionStorageService = new InMemoryStorageService();
+  const service = new DefaultPDPAddToCartService(localStorageService, sessionStorageService);
+
+  return {
+    service,
+    localStorageService,
+    sessionStorageService,
+    store
+  };
+}
+
+describe('testing pdp add to cart', () => {
+  const { service, localStorageService, sessionStorageService, store } =
+    preparePDPAddToCartService();
+
+  const result = createResultStub('Product 001');
```

Comment:
Build the stub with the url here with:

```suggestion
  const result = createResultStub('Product 001', {url: url});
```